### ISSUE TITLE
[Proposal] Add wrapper checker

### ISSUE THREAD

**Issue by pseudo-rnd-thoughts** [Farama-Foundation/Gymnasium#6]
In v24 / 25, we added an environment checker for users to check that environment follow the API correctly.
As some environments implement their own wrappers and to check our own, it would be helpful to add a wrapper checker.

### Proposed checks
* Check that all observations and actions spaces are valid (all observations are contained in space)
* Check the input and output types for `reset` and `step`

**Comment by Kallinteris-Andreas** [comment]
is this `passiveEnvChecker`?
close?

**Comment by pseudo-rnd-thoughts** [comment]
No, the passive environment checker is for the environment. I was more imagining a function similar to 'check_env' but for a wrapper

**Comment by noahfarr** [comment]
Is this still an open issue? If so, I'd like to give it a shot.

**Comment by pseudo-rnd-thoughts** [comment]
@noahfarr Hey, yeah, I don't think that anyone has worked on this so go for it

**Comment by CloseChoice** [comment]
Hi, 
I don't quite understand the task here. So do want to check the class of a wrapper or an instance of it? If we want to check the class, I assume that we don't really have enough information to check that all observations and spaces are valid. If we check an instance, so 

```python

env = make_env('some_env')
wrapped_env = Wrapper(env)
```

Then why not just apply `check_env` onto `wrapped_env`? What am I missing here?

**Comment by pseudo-rnd-thoughts** [comment]
This would be for checking an instance of a wrapper with an environment to check that it is correctly implemented
